versions/1.1/SPEC.md

```diff
@@ -0,0 +1,5775 @@
+      - [Map deserialization using read_map()](#map-deserialization-using-read_map)
+      - [Map deserialization using read_json()](#map-deserialization-using-read_json)
+    - [Pair](#pair-1)
+- [Appendix B: WDL Namespaces and Scopes](#appendix-b-wdl-namespaces-and-scopes)
+  - [Namespaces](#namespaces)
+  - [Scopes](#scopes)
+    - [Global Scope](#global-scope)
+    - [Task Scope](#task-scope)
+    - [Workflow Scope](#workflow-scope)
+    - [Cyclic References](#cyclic-references)
+    - [Namespaces without Scope](#namespaces-without-scope)
+  - [Evaluation Order](#evaluation-order)
+
+## Introduction
+
+WDL is meant to be a *human readable and writable* way to express tasks and workflows. Here is the "Hello World" tool in WDL:
+
+```wdl
+task hello {
+  input {
+    File infile
+    String pattern
+  }
+
+  command <<<
+    egrep '~{pattern}' '~{infile}'
+  >>>
+
+  runtime {
+    container: "my_image:latest"
+  }
+
+  output {
+    Array[String] matches = read_lines(stdout())
+  }
+}
+
+workflow wf {
+  input {
+    File infile
+    String pattern
+  }
+
+  call hello {
+    input: infile, pattern
+  }
+
+  output {
+    Array[String] matches = hello.matches
+  }
+}
+```
+
+This WDL document describes a task, called 'hello', and a workflow, called `wf`.
+
+* A `task` encapsulates a Bash script and a UNIX environment and presents them as a reusable function.
+* A `workflow` encapsulates a (directed, acyclic) graph of task calls that transforms input data to the desired outputs.
+
+Both workflows and tasks can accept input parameters and produce outputs. For example, the `wf` workflow has two input parameters, `File infile` and `String pattern`, and one output parameter, `Array[String] matches`. This simple workflow just passes its inputs to the `hello` task and returns its output.
+
+To execute this workflow, a WDL execution engine must be used (sometimes called the "WDL runtime" or "WDL implementation"). Some popular WDL execution engines are listed in the [README](https://github.com/openwdl/wdl#execution-engines). Along with the WDL file, the user must provide the implementation with values for the two input parameters. While implementations may provide their own mechanisms for launching workflows, all implementations minimally accept [inputs as JSON format](#json-input-format), which requires that inputs be fully qualified according to the namespacing rules described in the [Fully Qualified Names & Namespaced Identifiers](#fully-qualified-names--namespaced-identifiers) section. For example:
+
+|Variable     |Value    |
+|-------------|---------|
+|wf.pattern   |^[a-z]+$ |
+|wf.infile    |/file.txt|
+
+Or, in JSON format:
+
+```json
+{
+  "wf.pattern": "^[a-z]+$",
+  "wf.infile": "/file.txt"
+}
+```
+
+Running the `wf` workflow with these inputs would yield the following command line from the call to the `hello` task:
+
+```sh
+egrep '^[a-z]+$' '/file.txt'
+```
+
```

Comment:
A quick subheading here would help 
### Parallelizing Hello World

Reply:
Thanks - I reorganized the intro a bit to hopefully make it clearer that it's just meant as a quick example of what you can do with WDL.